libcontainer/process_linux.go

```diff
@@ -174,6 +175,42 @@ func (p *setnsProcess) start() (retErr error) {
 		case procHooks:
 			// This shouldn't happen.
 			panic("unexpected procHooks in setns")
+		case procSeccomp:
+			if p.config.Config.Seccomp.ListenerPath == "" {
+				return errors.New("listenerPath is not set")
+			}
+
+			seccompFd, err := recvSeccompFd(uintptr(p.pid()), uintptr(sync.Fd))
+			if err != nil {
+				return err
+			}
+			defer unix.Close(seccompFd)
+
+			bundle, annotations := utils.Annotations(p.config.Config.Labels)
+			containerProcessState := &specs.ContainerProcessState{
+				Version:  specs.Version,
+				Fds:      []string{specs.SeccompFdName},
+				Pid:      p.cmd.Process.Pid,
+				Metadata: p.config.Config.Seccomp.ListenerMetadata,
+				State: specs.State{
+					Version:     specs.Version,
+					ID:          p.config.ContainerId,
+					Status:      specs.StateRunning,
+					Pid:         p.initProcessPid,
+					Bundle:      bundle,
+					Annotations: annotations,
+				},
+			}
+			if err := sendContainerProcessState(p.config.Config.Seccomp.ListenerPath,
+				containerProcessState, seccompFd); err != nil {
+				return err
+			}
+
+			// Sync with child.
+			if err := writeSync(p.messageSockPair.parent, procSeccompDone); err != nil {
+				return fmt.Errorf("writing syncT 'SeccompDone': %w", err)
```

Comment:
It looks like you can move this error wrapping to `writeSyncWithFd`.

Reply:
Hmm, you mean `writeSync()`? In any case, no, I don't see how to move the error wrapping there.

We can do error wrapping there and keep `writeSync()` returning an error, but when calling a function here we _need_ to check if it returns an error or not, and act accordingly.

Am I missing something?

Reply:
Please see https://github.com/opencontainers/runc/pull/2682/commits/91ade3538df28dd920b5dc27cc9310d8c07f1922#r686185793